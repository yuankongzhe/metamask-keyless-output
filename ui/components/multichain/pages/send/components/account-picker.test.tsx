import React from 'react';
import configureStore from '../../../../../store/store';
import mockState from '../../../../../../test/data/mock-state.json';
import { fireEvent, renderWithProvider } from '../../../../../../test/jest';
import { SendPageAccountPicker } from '.';

const render = (props = {}) => {
  const store = configureStore({
    ...mockState,
    metamask: {
      ...mockState.metamask,
      permissionHistory: {
        'https://test.dapp': {
          eth_accounts: {
            accounts: {
              '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': 1596681857076,
            },
          },
        },
      },
      subjects: {
        'https://test.dapp': {
          permissions: {
            eth_accounts: {
              caveats: [
                {
                  type: 'restrictReturnedAccounts',
                  value: ['0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc'],
                },
              ],
              invoker: 'https://test.dapp',
              parentCapability: 'eth_accounts',
            },
          },
        },
      },
    },
    activeTab: {
      origin: 'https://test.dapp',
    },
  });
  return renderWithProvider(<SendPageAccountPicker {...props} />, store);
};

describe('SendPageAccountPicker', () => {
  describe('render', () => {
    it('renders correctly', () => {
      const { container, getByTestId } = render();
      expect(container).toMatchSnapshot();

      expect(getByTestId('send-page-account-picker')).toBeInTheDocument();
    });
  });

  describe('actions', () => {
    it('opens account picker when clicked', () => {
      const { getByTestId } = render();
      fireEvent.click(getByTestId('send-page-account-picker'));
      expect(
        document.querySelector('.multichain-account-menu-popover'),
      ).toBeInTheDocument();
    });
  });
});
