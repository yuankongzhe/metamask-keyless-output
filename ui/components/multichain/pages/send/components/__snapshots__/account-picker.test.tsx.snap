// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendPageAccountPicker render renders correctly 1`] = `
<div>
  <div
    class="mm-box mm-box--padding-bottom-6 mm-box--display-flex mm-box--flex-direction-column"
  >
    <label
      class="mm-box mm-text mm-label mm-text--body-md mm-text--font-weight-medium mm-box--padding-bottom-2 mm-box--display-inline-flex mm-box--align-items-center mm-box--color-text-default"
    >
      From
    </label>
    <button
      addressprops="[object Object]"
      class="mm-box mm-text mm-button-base mm-button-base--size-sm mm-button-base--block mm-button-base--ellipsis multichain-account-picker multichain-send-page__account-picker mm-text--body-md-medium mm-text--ellipsis mm-box--padding-0 mm-box--padding-top-4 mm-box--padding-right-4 mm-box--padding-bottom-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--gap-2 mm-box--justify-content-flex-start mm-box--align-items-center mm-box--width-full mm-box--color-text-default mm-box--background-color-transparent mm-box--rounded-lg mm-box--border-color-border-default mm-box--border-width-1 box--border-style-solid"
      data-testid="send-page-account-picker"
    >
      <span
        class="mm-box mm-text mm-text--inherit mm-text--ellipsis mm-box--display-flex mm-box--gap-2 mm-box--align-items-center mm-box--width-full mm-box--color-text-default"
      >
        <div
          class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-background-default box--border-style-solid box--border-width-1"
        >
          <div
            class="mm-avatar-account__jazzicon"
          >
            <div
              style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(250, 58, 0);"
            >
              <svg
                height="16"
                width="16"
                x="0"
                y="0"
              >
                <rect
                  fill="#18CDF2"
                  height="16"
                  transform="translate(-0.52419675189697 -1.6521420347302493) rotate(328.9 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#035E56"
                  height="16"
                  transform="translate(-9.149230854416022 5.2962309358743) rotate(176.2 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#F26602"
                  height="16"
                  transform="translate(8.333921009111961 -7.102569861498541) rotate(468.9 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
              </svg>
            </div>
          </div>
        </div>
        <span
          class="mm-box mm-text multichain-account-picker__label mm-text--body-md mm-text--ellipsis mm-box--padding-inline-start-2 mm-box--color-text-default"
          style="flex-grow: 1; text-align: start;"
        >
          Test Account
        </span>
      </span>
      <span
        class="mm-box mm-icon mm-icon--size-sm mm-box--margin-inline-start-1 mm-box--display-inline-block mm-box--color-icon-default"
        style="mask-image: url('./images/icons/arrow-down.svg');"
      />
    </button>
  </div>
</div>
`;
